### ISSUE TITLE
APIVersion 2019-06-01 is not available

### ISSUE THREAD

**Issue by internetionals** [ansible/ansible_collections_azure#136]
##### SUMMARY
When calling the `azure.azcollection.azure_rm_securitygroup` module, which used to work fine until recently, we now get the following error:

```
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: NotImplementedError: APIVersion 2019-06-01 is not available
```

Also tested using the current dev-branch of this project and updated requirements-pip.txt

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
`azure.azcollection.azure_rm_securitygroup`

##### ANSIBLE VERSION
```
ansible 2.9.9
  config file = /home/justin/co/platform/ansible/ansible.cfg
  configured module search path = [u'/home/justin/co/platform/ansible/library/modules', u'/home/justin/co/platform/ansible/common/library/modules']
  ansible python module location = /usr/local/lib/python2.7/dist-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.16 (default, Oct 10 2019, 22:02:15) [GCC 8.3.0]
```

##### CONFIGURATION
```
ANSIBLE_FORCE_COLOR(/home/justin/co/platform/ansible/ansible.cfg) = True
ANSIBLE_PIPELINING(/home/justin/co/platform/ansible/ansible.cfg) = True
ANSIBLE_SSH_ARGS(/home/justin/co/platform/ansible/ansible.cfg) = -o Compression=yes -o ForwardAgent={{ ssh_forward_agent|default(false)|bool|ternary('yes', 'no') }} -o ForwardX11=no -o GSSAPIDelegateCredentials=no -o ControlMaster=auto -o ControlPersist=60s -o UserKnownHo
ANSIBLE_SSH_CONTROL_PATH_DIR(/home/justin/co/platform/ansible/ansible.cfg) = ./var/run
BECOME_PLUGIN_PATH(/home/justin/co/platform/ansible/ansible.cfg) = [u'/home/justin/co/platform/ansible/library/plugins/become', u'/home/justin/co/platform/ansible/common/library/plugins/become']
COLLECTIONS_PATHS(/home/justin/co/platform/ansible/ansible.cfg) = [u'/home/justin/co/platform/ansible', u'/home/justin/co/platform/ansible/common']
DEFAULT_ACTION_PLUGIN_PATH(/home/justin/co/platform/ansible/ansible.cfg) = [u'/home/justin/co/platform/ansible/library/plugins/action', u'/home/justin/co/platform/ansible/common/library/plugins/action']
DEFAULT_ASK_PASS(/home/justin/co/platform/ansible/ansible.cfg) = False
DEFAULT_ASK_VAULT_PASS(/home/justin/co/platform/ansible/ansible.cfg) = False
DEFAULT_CACHE_PLUGIN_PATH(/home/justin/co/platform/ansible/ansible.cfg) = [u'/home/justin/co/platform/ansible/library/plugins/cache', u'/home/justin/co/platform/ansible/common/library/plugins/cache']
DEFAULT_CALLBACK_PLUGIN_PATH(/home/justin/co/platform/ansible/ansible.cfg) = [u'/home/justin/co/platform/ansible/library/plugins/callback', u'/home/justin/co/platform/ansible/common/library/plugins/callback']
DEFAULT_CONNECTION_PLUGIN_PATH(/home/justin/co/platform/ansible/ansible.cfg) = [u'/home/justin/co/platform/ansible/library/plugins/connection', u'/home/justin/co/platform/ansible/common/library/plugins/connection']
DEFAULT_FILTER_PLUGIN_PATH(/home/justin/co/platform/ansible/ansible.cfg) = [u'/home/justin/co/platform/ansible/library/plugins/filter', u'/home/justin/co/platform/ansible/common/library/plugins/filter']
DEFAULT_FORKS(/home/justin/co/platform/ansible/ansible.cfg) = 20
DEFAULT_GATHERING(/home/justin/co/platform/ansible/ansible.cfg) = explicit
DEFAULT_HASH_BEHAVIOUR(/home/justin/co/platform/ansible/ansible.cfg) = merge
DEFAULT_HOST_LIST(/home/justin/co/platform/ansible/ansible.cfg) = [u'/home/justin/co/platform/ansible/inventory']
DEFAULT_INVENTORY_PLUGIN_PATH(/home/justin/co/platform/ansible/ansible.cfg) = [u'/home/justin/co/platform/ansible/library/plugins/inventory', u'/home/justin/co/platform/ansible/common/library/plugins/inventory']
DEFAULT_LOAD_CALLBACK_PLUGINS(/home/justin/co/platform/ansible/ansible.cfg) = True
DEFAULT_LOOKUP_PLUGIN_PATH(/home/justin/co/platform/ansible/ansible.cfg) = [u'/home/justin/co/platform/ansible/library/plugins/lookup', u'/home/justin/co/platform/ansible/common/library/plugins/lookup']
DEFAULT_MODULE_PATH(/home/justin/co/platform/ansible/ansible.cfg) = [u'/home/justin/co/platform/ansible/library/modules', u'/home/justin/co/platform/ansible/common/library/modules']
DEFAULT_REMOTE_USER(/home/justin/co/platform/ansible/ansible.cfg) = root
DEFAULT_ROLES_PATH(/home/justin/co/platform/ansible/ansible.cfg) = [u'/home/justin/co/platform/ansible/roles', u'/home/justin/co/platform/ansible/common/roles']
DEFAULT_STDOUT_CALLBACK(/home/justin/co/platform/ansible/ansible.cfg) = yaml
DEFAULT_STRATEGY_PLUGIN_PATH(/home/justin/co/platform/ansible/ansible.cfg) = [u'/home/justin/co/platform/ansible/library/plugins/strategy', u'/home/justin/co/platform/ansible/common/library/plugins/strategy']
DEFAULT_TERMINAL_PLUGIN_PATH(/home/justin/co/platform/ansible/ansible.cfg) = [u'/home/justin/co/platform/ansible/library/plugins/terminal', u'/home/justin/co/platform/ansible/common/library/plugins/terminal']
DEFAULT_TEST_PLUGIN_PATH(/home/justin/co/platform/ansible/ansible.cfg) = [u'/home/justin/co/platform/ansible/library/plugins/test', u'/home/justin/co/platform/ansible/common/library/plugins/test']
DEFAULT_TIMEOUT(/home/justin/co/platform/ansible/ansible.cfg) = 15
DEFAULT_TRANSPORT(/home/justin/co/platform/ansible/ansible.cfg) = ssh
DEFAULT_VARS_PLUGIN_PATH(/home/justin/co/platform/ansible/ansible.cfg) = [u'/home/justin/co/platform/ansible/library/plugins/vars', u'/home/justin/co/platform/ansible/common/library/plugins/vars']
INJECT_FACTS_AS_VARS(/home/justin/co/platform/ansible/ansible.cfg) = False
INTERPRETER_PYTHON(/home/justin/co/platform/ansible/ansible.cfg) = /usr/bin/python
RETRY_FILES_SAVE_PATH(/home/justin/co/platform/ansible/ansible.cfg) = /home/justin/co/platform/ansible/var/retry_files
```

##### OS / ENVIRONMENT
Ansible Host: Debian Buster

##### STEPS TO REPRODUCE

Authenticated using `az login`

Playbook:

```yaml
- hosts: localhost
  tasks:
  - azure.azcollection.azure_rm_securitygroup:
      resource_group: doesnt_matter
      name: non_existent_security_group
      state: absent
```

##### EXPECTED RESULTS
Task gets executed normally

##### ACTUAL RESULTS
```
$ ansible-playbook pb.yml

PLAY [localhost] ***************************************************************************************************************************************************************************************************************************************************************

TASK [azure.azcollection.azure_rm_securitygroup] *******************************************************************************************************************************************************************************************************************************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: NotImplementedError: APIVersion 2019-06-01 is not available
fatal: [localhost]: FAILED! => changed=false
  module_stderr: |-
    Traceback (most recent call last):
      File "<stdin>", line 102, in <module>
      File "<stdin>", line 94, in _ansiballz_main
      File "<stdin>", line 40, in invoke_module
      File "/usr/lib/python2.7/runpy.py", line 188, in run_module
        fname, loader, pkg_name)
      File "/usr/lib/python2.7/runpy.py", line 82, in _run_module_code
        mod_name, mod_fname, mod_loader, pkg_name)
      File "/usr/lib/python2.7/runpy.py", line 72, in _run_code
        exec code in run_globals
      File "/tmp/ansible_azure.azcollection.azure_rm_securitygroup_payload_II2pLa/ansible_azure.azcollection.azure_rm_securitygroup_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_securitygroup.py", line 833, in <module>
      File "/tmp/ansible_azure.azcollection.azure_rm_securitygroup_payload_II2pLa/ansible_azure.azcollection.azure_rm_securitygroup_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_securitygroup.py", line 829, in main
      File "/tmp/ansible_azure.azcollection.azure_rm_securitygroup_payload_II2pLa/ansible_azure.azcollection.azure_rm_securitygroup_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_securitygroup.py", line 668, in __init__
      File "/tmp/ansible_azure.azcollection.azure_rm_securitygroup_payload_II2pLa/ansible_azure.azcollection.azure_rm_securitygroup_payload.zip/ansible_collections/azure/azcollection/plugins/module_utils/azure_rm_common.py", line 418, in __init__
      File "/tmp/ansible_azure.azcollection.azure_rm_securitygroup_payload_II2pLa/ansible_azure.azcollection.azure_rm_securitygroup_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_securitygroup.py", line 674, in exec_module
      File "/usr/local/lib/python2.7/dist-packages/azure/mgmt/network/network_management_client.py", line 1580, in network_security_groups
        raise NotImplementedError("APIVersion {} is not available".format(api_version))
    NotImplementedError: APIVersion 2019-06-01 is not available
  module_stdout: ''
  msg: |-
    MODULE FAILURE
    See stdout/stderr for the exact error
  rc: 1

PLAY RECAP *********************************************************************************************************************************************************************************************************************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0
```

**Comment by Fred-sun** [comment]
@internetionals  Thank you for taking the time to report this issue, this issue duplicate with previous issues #121 ! Can you try it in the way mentioned in #121? Thank you!

**Comment by internetionals** [comment]
Thanks for the pointer. It appears my Ansible prefers the global python lib over the user installed ones (I updated the `requirements-azure.txt` for my user, so as not to disturb other users until it was tested). So my `pip list` listed exactly the versions from the requirements file, but that version wasn't used.

I have now updated it system wide (`pip install --system -r requirements-azure.txt`) and it works again. Thanks @Fred-sun.
